lib/middleware.js

```diff
@@ -122,7 +122,7 @@ module.exports = function(options){
       if (typeof dest == 'string') {
         // check for dest-path overlap
         var overlap = compare(dest, path).length;
-        if (sep == path.charAt(0)) overlap++;
+        if ('/' == path.charAt(0)) overlap++;
```

Comment:
Path is a url path.  This will either be from the root `/path` or `/relative` path.  Am I missing anything?